src/js/handler/daygrid/resizeGuide.js

```diff
@@ -0,0 +1,147 @@
+ */
+'use strict';
+
+var util = require('tui-code-snippet');
+var config = require('../../config');
+var domutil = require('../../common/domutil');
+var datetime = require('../../common/datetime');
+var reqAnimFrame = require('../../common/reqAnimFrame');
+var TZDate = require('../../common/timezone').Date;
+
+/**
+ * @constructor
+ * @param {DayGridResize} resizeHandler - instance of DayGridResize
+ */
+function DayGridResizeGuide(resizeHandler) {
+    /**
+     * @type {DayGridResize}
+     */
+    this.resizeHandler = resizeHandler;
+
+    /**
+     * 실제로 이벤트 엘리먼트를 담는 엘리먼트
+     * @type {HTMLDIVElement}
+     */
+    this.scheduleContainer = null;
+
+    /**
+     * @type {function}
+     */
+    this.getScheduleDataFunc = null;
+
+    /**
+     * @type {HTMLDIVElement}
+     */
+    this.guideElement = null;
+
+    resizeHandler.on({
+        'dragstart': this._onDragStart,
+        'drag': this._onDrag,
+        'dragend': this._clearGuideElement,
+        'click': this._clearGuideElement
+    }, this);
+}
+
+/**
+ * Destroy method
+ */
+DayGridResizeGuide.prototype.destroy = function() {
+    this._clearGuideElement();
+    this.resizeHandler.off(this);
+    this.resizeHandler = this.scheduleContainer = this.getScheduleDataFunc =
+        this.guideElement = null;
+};
+
+/**
+ * Clear guide element.
+ */
+DayGridResizeGuide.prototype._clearGuideElement = function() {
+    domutil.remove(this.guideElement);
+
+    if (!util.browser.msie) {
+        domutil.removeClass(global.document.body, config.classname('resizing-x'));
+    }
+
+    this.getScheduleDataFunc = null;
+};
+
+/**
+ * Refresh guide element
+ * @param {number} newWidth - new width percentage value to resize guide element.
+ */
+DayGridResizeGuide.prototype.refreshGuideElement = function(newWidth) {
+    var guideElement = this.guideElement;
+
+    reqAnimFrame.requestAnimFrame(function() {
+        guideElement.style.width = newWidth + '%';
+    });
+};
+
+/**
+ * Return function that calculate guide element's new width percentage value.
+ * @param {object} dragStartEventData - dragstart schedule data.
+ * @returns {function} return function that calculate guide element new width percentage.
+ */
+DayGridResizeGuide.prototype.getGuideElementWidthFunc = function(dragStartEventData) {
+    var model = dragStartEventData.model,
+        viewOptions = this.resizeHandler.view.options,
+        fromLeft = (new TZDate(
+            model.start.getTime() - datetime.parse(viewOptions.renderStartDate)
+        )) / datetime.MILLISECONDS_PER_DAY | 0,
+        grids = dragStartEventData.grids;
+
+    return function(xIndex) {
+        var width = 0;
+        var i = 0;
+        var length = grids.length;
+        width += grids[fromLeft] ? grids[fromLeft].width : 0;
+
+        for (; i < length; i += 1) {
+            if (i > fromLeft && i <= xIndex) {
+                width += grids[i] ? grids[i].width : 0;
+            }
+        }
```

Comment:
this statement is more easy to ready => `for(i = fromLeft; i < length && i < xIndex; i += 1)`

Reply:
Ok. I will handle it at next refactoring.